src/test/java/org/springframework/data/repository/core/support/RepositoryMethodInvokerUnitTests.java

```diff
@@ -197,6 +197,7 @@ void capturesImperativeSuccessCorrectly() throws Exception {
 
 		assertThat(multicaster.first().getResult().getState()).isEqualTo(State.SUCCESS);
 		assertThat(multicaster.first().getResult().getError()).isNull();
+		assertThat(multicaster.first().getArguments()).isNotNull();
```

Comment:
I felt like its appropriate to check all state case results should have arguments present, any thoughts on this? I could make this a more explicit check if you want.